cloud/azure/datafactory/factories/mode/factoryusage.pm

```diff
@@ -0,0 +1,229 @@
+    foreach my $aggregation (@{$self->{az_aggregations}}) {
+        next if (!defined($aggregation) && !defined($self->{option_results}->{zeroed}));
+        $self->{metrics}->{$self->{az_resource}}->{statistics}->{lc($aggregation)}->{display} = lc($aggregation);
+        $self->{metrics}->{$self->{az_resource}}->{statistics}->{lc($aggregation)}->{timeframe} = $self->{az_timeframe};
+        foreach my $metric (@{$self->{az_metrics}}) {
+            my $metric_name = lc($metric);
+            $metric_name =~ s/ /_/g;
+            next if (!defined($metric_results{$self->{az_resource}}->{$metric_name}->{lc($aggregation)}) && !defined($self->{option_results}->{zeroed}));
+            $self->{metrics}->{$self->{az_resource}}->{statistics}->{lc($aggregation)}->{$metric} =
+                defined($metric_results{$self->{az_resource}}->{$metric_name}->{lc($aggregation)}) ?
+                $metric_results{$self->{az_resource}}->{$metric_name}->{lc($aggregation)} : 0;
+                $datas = 1;
+        }
+        # Compute percentages from metrics
+        next if (!defined($self->{metrics}->{$self->{az_resource}}->{statistics}->{lc($aggregation)}));
+        my $metricsAggregation = $self->{metrics}->{$self->{az_resource}}->{statistics}->{lc($aggregation)};
+        if (defined($metricsAggregation->{FactorySizeInGbUnits})) {
+            $metricsAggregation->{FactorySizeInGbUnits} = $metricsAggregation->{FactorySizeInGbUnits} * 1024;
+        }
+        if (defined($metricsAggregation->{MaxAllowedFactorySizeInGbUnits}) || defined($self->{option_results}->{zeroed})) {
+            $metricsAggregation->{MaxAllowedFactorySizeInGbUnits} = $metricsAggregation->{MaxAllowedFactorySizeInGbUnits} * 1024;
+            my $max_allowed_factory_size_in_gb_units = $metricsAggregation->{MaxAllowedFactorySizeInGbUnits};
+            if ($max_allowed_factory_size_in_gb_units > 0) {
+                $metricsAggregation->{factory_percentage_usage} = ($metricsAggregation->{FactorySizeInGbUnits} / $max_allowed_factory_size_in_gb_units) * 100;
+            } else {
+                $metricsAggregation->{factory_percentage_usage} = 0;
+            }
+            $datas = 1;
+        }
+        if (defined($metricsAggregation->{MaxAllowedResourceCount}) || defined($self->{option_results}->{zeroed})) {
+            my $max_allowed_resource_count = $metricsAggregation->{MaxAllowedResourceCount};
+            if ($max_allowed_resource_count > 0) {
+                $metricsAggregation->{resource_percentage_usage} = ($metricsAggregation->{ResourceCount} / $max_allowed_resource_count) * 100;
+            } else {
+                $metricsAggregation->{resource_percentage_usage} = 0;
+            }
+            $datas = 1;
+        }
+    }
+
+    if (!$datas) {
+        $self->{output}->add_option_msg(short_msg => 'No metrics. Check your options or use --zeroed option to set 0 on undefined values');
+        $self->{output}->option_exit();
+    }
```

Comment:
I think that it would be more aligned with other mode if you rely on the number of keys in $self->{metrics} .. But I could be wrong. 

Like this: 

```
    if (scalar(keys %{$self->{metrics}}) <= 0) {
        $self->{output}->add_option_msg(short_msg => "No metrics. Check your options or use --zeroed option to set 0 on undefined values");
        $self->{output}->option_exit();
    }
```

It will save you the $datas manipulation I guess